proof/refine/AARCH64/Detype_R.thy

```diff
@@ -1587,6 +1587,13 @@ lemma doMachineOp_modify:
   done
 
 context begin interpretation Arch . (*FIXME: arch_split*)
+
+lemma invs'_gsTypes_update:
+  "ksA' = ksArchState s \<Longrightarrow> invs' (s \<lparr>ksArchState := gsPTTypes_update f ksA'\<rparr>) = invs' s"
+  by (simp add: invs'_def valid_state'_def valid_global_refs'_def global_refs'_def
+                   valid_machine_state'_def valid_arch_state'_def
+           cong: option.case_cong)
```

Comment:
👍 to the move, but I don't think we should make it `simp`. It's written in this slightly strange form to make it match more than it should in `simp` (but it was needed that way in the goal, because the term there was just equivalent and not equal to `ksArchState s`)